### PR TITLE
Ngastelum/try catch blocks

### PR DESCRIPTION
**Pull request by ngastelum-ms**

### CODE REVIEW

File: iothub/device/src/IotHubDeviceClient.cs [hunk 1]
```diff
@@ -104,9 +107,11 @@ private IotHubDeviceClient(IotHubConnectionCredentials iotHubConnectionCredentia
         /// <seealso href="https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-file-upload#notify-iot-hub-of-a-completed-file-upload" />.
         /// <param name="notification">The notification details, including if the file upload succeeded.</param>
         /// <param name="cancellationToken">The cancellation token.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the operation has been canceled.</exception>
```
Comment (drwill-ms) [inline iothub/device/src/IotHubDeviceClient.cs:110]: Message copy/pasta error.

**Review by drwill-ms — APPROVED** [review]

File: iothub/device/src/IotHubModuleClient.cs [hunk 2]
```diff
@@ -157,17 +159,17 @@ public async Task SendEventAsync(string outputName, Message message, Cancellatio
...
 
-                Argument.AssertNotNullOrWhiteSpace(outputName, nameof(outputName));
+            var messagesList = messages?.ToList();
+            Argument.AssertNotNullOrEmpty(messagesList, nameof(messages));
 
-                var messagesList = messages?.ToList();
-                Argument.AssertNotNullOrEmpty(messagesList, nameof(messages));
+            try
+            {
+                ValidateModuleTransportHandler("SendEventBatchAsync for a named output");
```
Comment (andyk-ms) [inline iothub/device/src/IotHubModuleClient.cs:172]: see line 172 and 130, should both of them be consistently in or out of the try block?

**Review by andyk-ms — APPROVED** [review]
Assume comments are resolved.

File: iothub/device/src/IotHubModuleClient.cs (hunk 2 above)
Comment (drwill-ms) [reply iothub/device/src/IotHubModuleClient.cs:172]: Yep, this should be after line 165.

Comment (ngastelum-ms) [reply iothub/device/src/IotHubModuleClient.cs:172]: Moved both out of the try block

**Pull request merged**